src/chunk.c

```diff
@@ -884,27 +905,31 @@ chunk_fill_from_stub(Chunk *result, ChunkStub *chunk_stub, bool tuplock)
 
 	num_found = ts_scanner_scan(&ctx);
 
+	Assert(num_found == 0 || num_found == 1);
+
+	if (result->is_dropped)
+		return NULL;
+
 	if (num_found != 1)
 		elog(ERROR, "no chunk found with ID %d", chunk_stub->id);
 
-	if (NULL == result->cube)
-		result->cube = ts_hypercube_from_constraints(result->constraints, CurrentMemoryContext);
+	chunk = result->chunk;
+	Assert(NULL != chunk);
+	chunk->fd.id = chunk_stub->id;
+	chunk->constraints = chunk_stub->constraints;
+	chunk->cube = chunk_stub->cube;
+
+	if (NULL == chunk->cube)
+		chunk->cube = ts_hypercube_from_constraints(chunk->constraints, CurrentMemoryContext);
 	else
```

Comment:
I know that we don't typically use braces for single lines of code, but given the comment and I think they would improve readability here.  At the very least can you remove the empty line.